### DISCUSSION TITLE
Splitting assets and files permissions?

### DISCUSSION THREAD

**Discussion by infomiho** [directus/directus#15690] [Feature Requests]
Hey good people of Directus,
what do you think about decoupling `/assets` access from `/files` access? 🤔 

One scenario that doesn't seem all that safe is: **if you want to serve images through Directus, everyone can browse all your files via `/file`**

**Comment by azrikahar** [comment]
> if you want to serve images through Directus, everyone can browse all your files via `/files`

Assuming by everyone, you meant Public Role was given full read permission to `directus_files`, you should be able to restrict or limit it to a particular folder so that not all files are exposed:

![image](https://user-images.githubusercontent.com/42867097/193985612-4b1e2d14-77da-4184-ad80-7045171e74fb.png)

With that said, the Public Role can still indeed make a GET request to `/files` to get a list of files _within that folder_, so that folder should only contain files that are meant to be made public. On a related note, there's a discussion on a potential List permission in #11464 👍

**Reply by yondr-agency-15** [reply]
Can be found here: 
![image](https://user-images.githubusercontent.com/93581121/215057230-3b06b640-9dc1-4373-8c0e-bd3a7f563990.png)

**Reply by ricricucit** [reply]
will this config be kept in the `yaml`, when running `npx directus schema snapshot`?
I cannot understand that, by reading the generated YAML file.

**Reply by ricricucit** [reply]
✅  answering myself: this kind of stuff is (of course!) on DB level.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
